tests/map_js_test.go

```diff
@@ -0,0 +1,117 @@
+//go:build js && !wasm
```

Comment:
General test-related nitpick: unlike error messages, test errors are full sentences, starting with a capital letter and ending with punctuation.

Also, this is my personal preference, but I find that phrasing test errors in "Got: $some_unexpected_result. Want: $expected_behavior" pattern makes it easier to debug failures of the test you may not be familiar with. The human-friendly message in the "want" part gives the opportunity to convey finer semantics than, for example, "message doesn't contain a particular substring". (See my comment below for an example).

Since this is my personal preference rather than a formally adopted style requirement, you don't have to consider it merge-blocking, but hopefully the reasoning makes sense to you :)